### PR TITLE
Switch to use github action to check links in markdown

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

* [x] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

### CODE REVIEW

**Review by bwplotka — APPROVED** [review]
So.. all our links works? 

Suspicious (:

File: .bingo/Variables.mk [hunk 1]
```diff
@@ -88,12 +88,6 @@ $(JSONNETFMT): .bingo/jsonnetfmt.mod
 	@echo "(re)installing $(GOBIN)/jsonnetfmt-v0.16.0"
 	@cd .bingo && $(GO) build -modfile=jsonnetfmt.mod -o=$(GOBIN)/jsonnetfmt-v0.16.0 "github.com/google/go-jsonnet/cmd/jsonnetfmt"
 
-LICHE := $(GOBIN)/liche-v0.0.0-20200229003944-f57a5d1c5be4
```
Comment (bwplotka) [inline .bingo/Variables.mk:91]: You need to also remove `.bingo/liche.mod` to make it permanent.

Comment (daixiang0) [reply .bingo/Variables.mk:91]: Seems you need enable new github action in pr, details refer to [this](https://docs.github.com/en/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository), there is not a github action named as "Check Markdown links".

Comment (bwplotka) [reply .bingo/Variables.mk:91]: ack

Comment (bwplotka) [reply .bingo/Variables.mk:91]: It looks like it's enabled:

image

Comment (daixiang0) [reply .bingo/Variables.mk:91]: Now all work!

**Comment by daixiang0** [comment]
@bwplotka any chance to merge it?

File: CHANGELOG.md [hunk 2]
```diff
@@ -80,6 +80,7 @@ Thanos Rule's `/api/v1/rules` endpoint no longer returns the old, deprecated `pa
 sse_config:
   type: SSE-S3
 ```
+- [#3106](https://github.com/thanos-io/thanos/pull/3106) Switch to use github action to check links in markdown.
```
Comment (bwplotka) [inline CHANGELOG.md:83]: Definitely no need for changlog item - not user impacting change.

**Review by bwplotka — APPROVED** [review]
Thanks, just one thing and LGTM (: 

I feel like we need something more customized in future, but it's perfect for now!

**Comment by daixiang0** [comment]
ping @thanos-io/thanos-maintainers

**Pull request merged**